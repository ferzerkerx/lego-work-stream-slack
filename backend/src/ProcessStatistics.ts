export class ProcessStatistics {
  static triggers: number = 0;
  static convos: number = 0;

  static formatUptime(uptime: number): string {
    let unit: string = 'second';
    if (uptime > 60) {
      uptime = uptime / 60;
      unit = 'minute';
    }
    if (uptime > 60) {
      uptime = uptime / 60;
      unit = 'hour';
    }
    if (uptime != 1) {
      unit = unit + 's';
    }
    return `${uptime} ${unit}`;
  }
}
