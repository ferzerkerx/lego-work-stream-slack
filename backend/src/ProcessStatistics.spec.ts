import { ProcessStatistics } from './ProcessStatistics';

describe('ProcessStatistics', () => {
  test('formatUptime', () => {
    ProcessStatistics.triggers = 10;
    ProcessStatistics.convos = 40;

    const value = ProcessStatistics.formatUptime(100);
    expect(value).toMatchSnapshot();
  });
});
