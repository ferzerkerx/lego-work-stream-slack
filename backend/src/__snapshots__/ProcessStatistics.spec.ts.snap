// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessStatistics formatUptime 1`] = `"1.6666666666666667 minutes"`;
